pkg/frontend/querymiddleware/codec.go

```diff
@@ -226,6 +230,19 @@ func decodeOptions(r *http.Request, opts *Options) {
 	}
 }
 
+func decodeInstantQueryOptions(r *http.Request, opts *Options) {
+	for _, value := range r.Header.Values(instantSplitIntervalControlHeader) {
+		splitInterval, err := time.ParseDuration(value)
+		if err != nil {
+			break
```

Comment:
I think should be a continue. Same change should be done above (in existing code). I don't see a reason to stop parsing other values for the same header, if only 1 breaks.

Reply:
Yes, good point 👍